pkg/statsd/dispatcher.go

```diff
@@ -48,30 +52,61 @@ func NewMetricDispatcher(numWorkers int, perWorkerBufferSize int, af AggregatorF
 	}
 }
 
-// Run runs the MetricDispatcher.
-func (d *MetricDispatcher) Run(ctx context.Context, done gostatsd.Done) {
```

Comment:
If it's launched as a go routine, then we start pumping metrics in to it before it's got running=true.  If we move the flag to NewMetricDispatcher, then we could be accepting metrics on something we haven't started, which seems like accepting metrics on something we've stopped.